mettle/api/api.go

```diff
@@ -202,6 +203,7 @@ type server struct {
 	mutex            sync.Mutex
 	numPollers       int
 	deleteJobsTicker *time.Ticker
+	actuallyValidate bool
```

Comment:
Is it necessary to add an extra field on the server struct here? Could we just check the length of s.platform below?

Reply:
+1, also it almost feels like the check should even be part of `validatePlatform`.

Reply:
I refactored in this way mainly because I was initially passing through a flag to ignore, rather than the approach I took in the end.

Personally, I prefer to read it this way, but not 100% wedded to it, can just in line the check.

Reply:
The other thing is that we expect `Execute` (and therefore validatePlatform) to be called a whole bunch of times, but the platform config itself never actually changes, and I think the boolean on the struct kinda indicates that more clearly.

I do agree that it should be pushed down into `validatePlatform` actually, I think that keeps the code clearer.

Reply:
This makes sense to me. Thanks Xander